### PR TITLE
Update bundles for latest releases and multiple architectures

### PR DESCRIPTION
**Pull request by ryan-beisner**
Update bundles to represent the current releases of OpenStack Charms, Ubuntu OpenStack, Juju and MAAS behavior.  This includes a fairly major reorganization of the bundles and directories for use in automation.  The development bundles each have a separate directory (ubuntu-openstack) for the purpose of continuous testing, whereas the stable bundles (base, telemetry, lxd) only ever have one maintained version (latest stable release).

The helper scripts are now pulled in from lp:openstack-charm-testing, as the copies here had bit-rotted.  Tox and travis will check to make sure with each PR that those tools are in sync.

### CODE REVIEW

File: development/openstack-base-trusty-mitaka/README.md [hunk 1]
```diff
@@ -57,42 +57,42 @@ All commands are executed from within the expanded bundle.
...
 
 In order to run instances on your cloud, you'll need to upload an image to boot instances from:
 
-    mkdir -p ~/images
-    wget -O ~/images/trusty-server-cloudimg-amd64-disk1.img \
-        http://cloud-images.ubuntu.com/trusty/current/trusty-server-cloudimg-amd64-disk1.img
-    glance image-create --name="trusty" --visibility public --progress \
-        --container-format=bare --disk-format=qcow2 \
-        < ~/images/trusty-server-cloudimg-amd64-disk1.img
+    curl http://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-disk1.img | \
```
Comment (andrewdmcleod) [inline development/openstack-base-trusty-mitaka/README.md]: is it worth providing links to other images for other arches here?

File: development/openstack-base-trusty-mitaka/README.md [hunk 2]
```diff
@@ -57,42 +57,42 @@ All commands are executed from within the expanded bundle.
...
 
 for example (for a private cloud):
 
-    ./neutron-ext-net -g 10.230.168.1 -c 10.230.168.0/21 \
+    ./neutron-ext-net --network-type flat
+        -g 10.230.168.1 -c 10.230.168.0/21 \
         -f 10.230.168.10:10.230.175.254 ext_net
 
-You'll need to adapt the parameters for the network configuration which eth1 on all the servers is connected to; in a public cloud deployment these ports would be connected to a publicly-addressable part of the Internet.
+You'll need to adapt the parameters for the network configuration that eth1 on all the servers is connected to; in a public cloud deployment these ports would be connected to a publicly addressable part of the Internet.
```
Comment (andrewdmcleod) [inline development/openstack-base-trusty-mitaka/README.md]: eth1/eno1/??

File: development/openstack-base-xenial-mitaka/README.md [hunk 3]
```diff
@@ -57,42 +57,42 @@ All commands are executed from within the expanded bundle.
...
 
 for example (for a private cloud):
 
-    ./neutron-ext-net -g 10.230.168.1 -c 10.230.168.0/21 \
+    ./neutron-ext-net --network-type flat
+        -g 10.230.168.1 -c 10.230.168.0/21 \
         -f 10.230.168.10:10.230.175.254 ext_net
 
-You'll need to adapt the parameters for the network configuration which eth1 on all the servers is connected to; in a public cloud deployment these ports would be connected to a publicly-addressable part of the Internet.
+You'll need to adapt the parameters for the network configuration that eth1 on all the servers is connected to; in a public cloud deployment these ports would be connected to a publicly addressable part of the Internet.
```
Comment (andrewdmcleod) [inline development/openstack-base-xenial-mitaka/README.md]: eth/eno/???

File: development/openstack-base-xenial-newton/README.md [hunk 4]
```diff
@@ -57,42 +57,42 @@ All commands are executed from within the expanded bundle.
...
 
 for example (for a private cloud):
 
-    ./neutron-ext-net -g 10.230.168.1 -c 10.230.168.0/21 \
+    ./neutron-ext-net --network-type flat
+        -g 10.230.168.1 -c 10.230.168.0/21 \
         -f 10.230.168.10:10.230.175.254 ext_net
 
-You'll need to adapt the parameters for the network configuration which eth1 on all the servers is connected to; in a public cloud deployment these ports would be connected to a publicly-addressable part of the Internet.
+You'll need to adapt the parameters for the network configuration that eth1 on all the servers is connected to; in a public cloud deployment these ports would be connected to a publicly addressable part of the Internet.
```
Comment (andrewdmcleod) [inline development/openstack-base-xenial-newton/README.md]: eth/eno

File: development/openstack-base-xenial-ocata/README.md [hunk 5]
```diff
@@ -57,42 +57,42 @@ All commands are executed from within the expanded bundle.
...
 
 for example (for a private cloud):
 
-    ./neutron-ext-net -g 10.230.168.1 -c 10.230.168.0/21 \
+    ./neutron-ext-net --network-type flat
+        -g 10.230.168.1 -c 10.230.168.0/21 \
         -f 10.230.168.10:10.230.175.254 ext_net
 
-You'll need to adapt the parameters for the network configuration which eth1 on all the servers is connected to; in a public cloud deployment these ports would be connected to a publicly-addressable part of the Internet.
+You'll need to adapt the parameters for the network configuration that eth1 on all the servers is connected to; in a public cloud deployment these ports would be connected to a publicly addressable part of the Internet.
```
Comment (andrewdmcleod) [inline development/openstack-base-xenial-ocata/README.md]: eth/eno

**Comment by andrewdmcleod** [comment]
minor comments, otherwise lgtm

**Comment by ryan-beisner** [comment]
### amd64 openstack-base xenial-pike stable deploy OK

```
(openstack-client) 2 jenkins@juju-0d10ff-osci-4:~/temp/openstack-charm-testing⟫ ping -c 2 10.245.171.11
PING 10.245.171.11 (10.245.171.11) 56(84) bytes of data.
64 bytes from 10.245.171.11: icmp_seq=1 ttl=61 time=1.39 ms
64 bytes from 10.245.171.11: icmp_seq=2 ttl=61 time=1.19 ms

--- 10.245.171.11 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1004ms
rtt min/avg/max/mdev = 1.198/1.295/1.393/0.103 ms
(openstack-client) jenkins@juju-0d10ff-osci-4:~/temp/openstack-charm-testing⟫ ssh -i ~/testkey.pem ubuntu@10.245.171.11 uname -a
The authenticity of host '10.245.171.11 (10.245.171.11)' can't be established.
ECDSA key fingerprint is SHA256:zzZ932Ky3SCyE48E8q2ZNzzBC9UvWGLUesiWQIumLho.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '10.245.171.11' (ECDSA) to the list of known hosts.
Linux xenial-161247-2 4.4.0-93-generic #116-Ubuntu SMP Fri Aug 11 21:17:51 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

(openstack-client) jenkins@juju-0d10ff-osci-4:~/temp/openstack-bundles/stable/openstack-base⟫ openstack server list
+--------------------------------------+-----------------+--------+-----------------------------------+--------+----------+
| ID                                   | Name            | Status | Networks                          | Image  | Flavor   |
+--------------------------------------+-----------------+--------+-----------------------------------+--------+----------+
| 681697a8-6e8c-4069-b2f8-031386ad94af | xenial-161247-3 | ACTIVE | private=172.16.0.9, 10.245.171.0  | xenial | m1.small |
| 8d40e10b-30a8-41f4-afbe-21d185099f2a | xenial-161247-2 | ACTIVE | private=172.16.0.7, 10.245.171.11 | xenial | m1.small |
| 9c624edf-1261-4c3f-862c-3c931f451927 | xenial-161247-1 | ACTIVE | private=172.16.0.4, 10.245.171.3  | xenial | m1.small |
+--------------------------------------+-----------------+--------+-----------------------------------+--------+----------+
```

```
Model           Controller      Cloud/Region  Version  SLA
auto-osci-lb01  auto-osci-lb01  ruxton-maas   2.2.2    unsupported

App                    Version       Status  Scale  Charm                  Store       Rev  OS      Notes
ceph-mon               12.2.0        active      3  ceph-mon               jujucharms   12  ubuntu  
ceph-osd               12.2.0        active      3  ceph-osd               jujucharms  246  ubuntu  
ceph-radosgw           12.2.0        active      1  ceph-radosgw           jujucharms  251  ubuntu  
cinder                 11.0.0        active      1  cinder                 jujucharms  264  ubuntu  
cinder-ceph            11.0.0        active      1  cinder-ceph            jujucharms  227  ubuntu  
glance                 15.0.0        active      1  glance                 jujucharms  259  ubuntu  
keystone               12.0.0        active      1  keystone               jujucharms  268  ubuntu  
mysql                  5.6.34-26.19  active      1  percona-cluster        jujucharms  254  ubuntu  
neutron-api            11.0.0        active      1  neutron-api            jujucharms  252  ubuntu  
neutron-gateway        11.0.0        active      1  neutron-gateway        jujucharms  238  ubuntu  
neutron-openvswitch    11.0.0        active      3  neutron-openvswitch    jujucharms  244  ubuntu  
nova-cloud-controller  16.0.0        active      1  nova-cloud-controller  jujucharms  300  ubuntu  
nova-compute           16.0.0        active      3  nova-compute           jujucharms  273  ubuntu  
ntp                    4.2.8p4+dfsg  active      4  ntp                    jujucharms   18  ubuntu  
openstack-dashboard    12.0.0        active      1  openstack-dashboard    jujucharms  250  ubuntu  
rabbitmq-server        3.5.7         active      1  rabbitmq-server        jujucharms   65  ubuntu  

Unit                      Workload  Agent      Machine  Public address  Ports              Message
ceph-mon/0                active    idle       1/lxd/0  10.245.168.24                      Unit is ready and clustered
ceph-mon/1*               active    idle       2/lxd/0  10.245.168.44                      Unit is ready and clustered
ceph-mon/2                active    idle       3/lxd/0  10.245.168.30                      Unit is ready and clustered
ceph-osd/0                active    idle       1        10.245.168.41                      Unit is ready (1 OSD)
ceph-osd/1*               active    idle       2        10.245.168.49                      Unit is ready (1 OSD)
ceph-osd/2                active    idle       3        10.245.168.39                      Unit is ready (1 OSD)
ceph-radosgw/0*           active    idle       0/lxd/0  10.245.168.23   80/tcp             Unit is ready
cinder/0*                 active    idle       1/lxd/1  10.245.168.28   8776/tcp           Unit is ready
  cinder-ceph/0*          active    idle                10.245.168.28                      Unit is ready
glance/0*                 active    idle       2/lxd/1  10.245.168.45   9292/tcp           Unit is ready
keystone/0*               active    idle       3/lxd/1  10.245.168.31   5000/tcp           Unit is ready
mysql/0*                  active    idle       0/lxd/1  10.245.168.25   3306/tcp           Unit is ready
neutron-api/0*            active    idle       1/lxd/2  10.245.168.29   9696/tcp           Unit is ready
neutron-gateway/0*        active    idle       0        10.245.168.40                      Unit is ready
  ntp/3                   active    idle                10.245.168.40   123/udp            Unit is ready
nova-cloud-controller/0*  active    idle       2/lxd/2  10.245.168.46   8774/tcp,8778/tcp  Unit is ready
nova-compute/0            active    idle       1        10.245.168.41                      Unit is ready
  neutron-openvswitch/2   active    idle                10.245.168.41                      Unit is ready
  ntp/2                   active    idle                10.245.168.41   123/udp            Unit is ready
nova-compute/1*           active    executing  2        10.245.168.49                      (update-status) Unit is ready
  neutron-openvswitch/1   active    idle                10.245.168.49                      Unit is ready
  ntp/1                   active    idle                10.245.168.49   123/udp            Unit is ready
nova-compute/2            active    idle       3        10.245.168.39                      Unit is ready
  neutron-openvswitch/0*  active    idle                10.245.168.39                      Unit is ready
  ntp/0*                  active    idle                10.245.168.39   123/udp            Unit is ready
openstack-dashboard/0*    active    idle       3/lxd/2  10.245.168.27   80/tcp,443/tcp     Unit is ready
rabbitmq-server/0*        active    executing  0/lxd/2  10.245.168.26   5672/tcp           (update-status) Unit is ready

Machine  State    DNS            Inst id              Series  AZ       Message
0        started  10.245.168.40  ya8ymh               xenial  default  Deployed
0/lxd/0  started  10.245.168.23  juju-f42e5a-0-lxd-0  xenial           Container started
0/lxd/1  started  10.245.168.25  juju-f42e5a-0-lxd-1  xenial           Container started
0/lxd/2  started  10.245.168.26  juju-f42e5a-0-lxd-2  xenial           Container started
1        started  10.245.168.41  cbxwat               xenial  default  Deployed
1/lxd/0  started  10.245.168.24  juju-f42e5a-1-lxd-0  xenial           Container started
1/lxd/1  started  10.245.168.28  juju-f42e5a-1-lxd-1  xenial           Container started
1/lxd/2  started  10.245.168.29  juju-f42e5a-1-lxd-2  xenial           Container started
2        started  10.245.168.49  tmrm8d               xenial  default  Deployed
2/lxd/0  started  10.245.168.44  juju-f42e5a-2-lxd-0  xenial           Container started
2/lxd/1  started  10.245.168.45  juju-f42e5a-2-lxd-1  xenial           Container started
2/lxd/2  started  10.245.168.46  juju-f42e5a-2-lxd-2  xenial           Container started
3        started  10.245.168.39  66wnqf               xenial  default  Deployed
3/lxd/0  started  10.245.168.30  juju-f42e5a-3-lxd-0  xenial           Container started
3/lxd/1  started  10.245.168.31  juju-f42e5a-3-lxd-1  xenial           Container started
3/lxd/2  started  10.245.168.27  juju-f42e5a-3-lxd-2  xenial           Container started

Relation                 Provides               Consumes               Type
mon                      ceph-mon               ceph-mon               peer
mon                      ceph-mon               ceph-osd               regular
mon                      ceph-mon               ceph-radosgw           regular
ceph                     ceph-mon               cinder-ceph            regular
ceph                     ceph-mon               glance                 regular
ceph                     ceph-mon               nova-compute           regular
cluster                  ceph-radosgw           ceph-radosgw           peer
identity-service         ceph-radosgw           keystone               regular
cluster                  cinder                 cinder                 peer
storage-backend          cinder                 cinder-ceph            subordinate
image-service            cinder                 glance                 regular
identity-service         cinder                 keystone               regular
shared-db                cinder                 mysql                  regular
cinder-volume-service    cinder                 nova-cloud-controller  regular
amqp                     cinder                 rabbitmq-server        regular
cluster                  glance                 glance                 peer
identity-service         glance                 keystone               regular
shared-db                glance                 mysql                  regular
image-service            glance                 nova-cloud-controller  regular
image-service            glance                 nova-compute           regular
amqp                     glance                 rabbitmq-server        regular
cluster                  keystone               keystone               peer
shared-db                keystone               mysql                  regular
identity-service         keystone               neutron-api            regular
identity-service         keystone               nova-cloud-controller  regular
identity-service         keystone               openstack-dashboard    regular
cluster                  mysql                  mysql                  peer
shared-db                mysql                  neutron-api            regular
shared-db                mysql                  nova-cloud-controller  regular
cluster                  neutron-api            neutron-api            peer
neutron-plugin-api       neutron-api            neutron-gateway        regular
neutron-plugin-api       neutron-api            neutron-openvswitch    regular
neutron-api              neutron-api            nova-cloud-controller  regular
amqp                     neutron-api            rabbitmq-server        regular
cluster                  neutron-gateway        neutron-gateway        peer
quantum-network-service  neutron-gateway        nova-cloud-controller  regular
juju-info                neutron-gateway        ntp                    subordinate
amqp                     neutron-gateway        rabbitmq-server        regular
neutron-plugin           neutron-openvswitch    nova-compute           regular
amqp                     neutron-openvswitch    rabbitmq-server        regular
cluster                  nova-cloud-controller  nova-cloud-controller  peer
cloud-compute            nova-cloud-controller  nova-compute           regular
amqp                     nova-cloud-controller  rabbitmq-server        regular
neutron-plugin           nova-compute           neutron-openvswitch    subordinate
compute-peer             nova-compute           nova-compute           peer
juju-info                nova-compute           ntp                    subordinate
amqp                     nova-compute           rabbitmq-server        regular
ntp-peers                ntp                    ntp                    peer
cluster                  openstack-dashboard    openstack-dashboard    peer
cluster                  rabbitmq-server        rabbitmq-server        peer
```

File: development/openstack-base-trusty-mitaka/README.md (hunk 1 above)
Comment (ryan-beisner) [reply development/openstack-base-trusty-mitaka/README.md]: Yes, in a separate review IMO.

**Comment by andrewdmcleod** [comment]
go go gadget merge

**Pull request merged**